### PR TITLE
feat(lido): add stETH vault flow assertion suite

### PR DESCRIPTION
**Pull request by makemake-kbo**
## Lido stETH vault flow assertion suite

Generic Credible Layer assertions for **any vault that custodies stETH/wstETH and deploys it into an Aave-style market** (Veda BoringVault / GGV, Mellow flexible-vaults / EarnETH, Lido V3 stVaults, or comparable loopers). Everything is expressed at transaction boundaries, configured with addresses and thresholds — no vault-stack-specific coupling.

### Assertions

| Contract | What it enforces |
|---|---|
| `LidoStEthExitBufferAssertion` | **Withdrawable-stETH buffer floor** on every transaction (idle stETH-equivalent stays above the larger of an absolute minimum and a fraction of total exposure → the vault is never fully deployed) plus a **rolling-window outflow circuit breaker** on stETH/wstETH. The breaker runs in the block builder, so a burst drain is never included even against a fully compromised signer pipeline. |
| `LidoStEthVaultRiskAssertion` | Reduce-only regime under unhealthy / off-peg / illiquid conditions, exit-liquidity and collateral-withdrawability guards, and a health-factor + collateral-ratio position envelope. |
| `LidoStEthVaultPegAssertion` | Supply-change depeg gate (any mint/burn reverts while stETH/ETH is outside the peg band) and wstETH rate integrity. |
| `LidoStEthVaultNavAssertion` | Reported share rate checked against NAV recomputed from on-chain state, in both directions. |

Shared `LidoVaultHelpers` / `LidoVaultInterfaces` back all four.

### Honest boundaries

- **Requestable, not claimable.** On-chain we can guarantee stETH stays *requestable* to Lido's withdrawal queue at any time; instant claimability is bounded by the validator exit queue and the 1,000 ETH/request cap, which no per-vault check controls.
- **Outflow breaker is destination-blind in v1** — it rate-limits all stETH/wstETH leaving the vault; a destination-aware variant (exempt transfers to Lido's `WithdrawalQueueERC721`) is the named next step.
- No protection against market-price loss; third-party Aave liquidations are out of reach; Lido core is untouched.

### Build

```sh
FOUNDRY_PROFILE=lido forge build
```

Builds clean; `forge fmt --check` passes.

Closes ENG-3428.

### CODE REVIEW

File: examples/lido/src/LidoStEthVaultRiskAssertion.sol [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+        uint256 minCollateralLiquidityBps;
+    }
+
+    constructor(RiskConfig memory config) {
+        require(config.vault != address(0), "LidoVault: zero vault");
+        require(config.aavePool != address(0), "LidoVault: zero pool");
+        require(config.minHealthFactor >= 1e18, "LidoVault: floor below liquidation");
+        require(config.reduceOnlyHealthFactor >= config.minHealthFactor, "LidoVault: band below floor");
+        require(
+            config.minExitLiquidityBps == 0
```
Comment (mateo-mro) [inline examples/lido/src/LidoStEthVaultRiskAssertion.sol]: The scoped formatter check fails on this block. The PR body says formatting passes, but this changed file still needs forge fmt.

File: examples/lido/README.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+- LidoStEthExitBufferAssertion.sol
+- LidoStEthVaultRiskAssertion.sol
+- LidoStEthVaultPegAssertion.sol
+- LidoStEthVaultNavAssertion.sol
+- LidoVaultHelpers.sol
+- LidoVaultInterfaces.sol
+
+## Next steps
+
+- `CredibleTest` E2E coverage (mock pool/rate-source/feed/token knobs that trip one invariant at a time)
```
Comment (mateo-mro) [inline examples/lido/README.md]: I would not leave this as a next step in the PR. These are executable assertions with threshold and oracle edge cases, so the examples should land with CredibleTest coverage for each invariant.

File: examples/lido/src/LidoVaultHelpers.sol [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+            return false;
+        }
+
+        PhEvm.StaticCallResult memory result =
+            ph.staticcallAt(feed, abi.encodeCall(IChainlinkFeedLike.latestRoundData, ()), FORK_VIEW_GAS, fork);
+        if (!result.ok) {
+            return true;
+        }
+
+        (, int256 answer,,,) = abi.decode(result.data, (uint80, int256, uint256, uint256, uint80));
```
Comment (mateo-mro) [inline examples/lido/src/LidoVaultHelpers.sol]: latestRoundData also returns updatedAt and answeredInRound. This should fail closed on stale or incomplete rounds, probably with a configured max age, otherwise an old peg price can keep the risk and supply change gates open during an oracle outage.

**Review by mateo-mro — CHANGES_REQUESTED** [review]

File: examples/lido/src/LidoStEthVaultRiskAssertion.sol (hunk 1 above)
Comment (makemake-kbo) [reply examples/lido/src/LidoStEthVaultRiskAssertion.sol]: Fixed in 29c0f58. `forge fmt --check` passes on this file with my local toolchain (forge 1.5.1 / solc 0.8.30), so the failure was a forge-fmt version disagreeing on this deeply-nested boolean. Rather than chase a version, I removed the version-sensitive construct: the `minExitLiquidityBps` / `minCollateralLiquidityBps` validations are now `if`-guarded, one short `require` per token (no line wrapping for any formatter to disagree on), which also yields more precise errors (`zero borrowed asset`, `zero collateral supply token`, etc.).

File: examples/lido/README.md (hunk 2 above)
Comment (makemake-kbo) [reply examples/lido/README.md]: Done in 29c0f58 — dropped the next-step and landed coverage. Added `examples/lido/test/` with 59 passing CredibleTest tests (one pass + one trip per invariant) plus the oracle/threshold edge cases: depeg both directions, stale / incomplete / carried-over feed rounds, wstETH rate-decrease & provider-desync, buffer floor (relative + absolute + wstETH-at-Lido-rate), exit-liquidity, collateral-withdrawability, the full health-factor envelope, and NAV above/below tolerance + insolvency. Shared mocks live in `test/LidoMocks.sol`. Run with `FOUNDRY_PROFILE=lido pcl test`. The `watchCumulativeOutflow` breaker isn't simulated by local `pcl test` (executor-driven), so its hard-revert is exercised directly and noted, consistent with the Lighter example.

File: examples/lido/src/LidoVaultHelpers.sol (hunk 3 above)
Comment (makemake-kbo) [reply examples/lido/src/LidoVaultHelpers.sol]: Fixed in 29c0f58. `_isStEthDepeggedAt` now decodes `updatedAt` and `answeredInRound` and fails closed on: a non-positive answer, an incomplete round (`updatedAt == 0`), a carried-over answer (`answeredInRound < roundId`), and an answer older than a configurable `maxFeedStalenessSecs` (zero disables only the age bound; the round-integrity checks always apply). I threaded `maxFeedStalenessSecs` through both consumers (the risk `RiskConfig` and the peg constructor), so an oracle outage can no longer hold the risk-allocation or share mint/burn gates open on a stale on-peg price. Covered by `testMintWhileFeedStaleTrips` / `testMintWhileRoundIncompleteTrips` / `testMintWhileRoundStaleTrips`, and documented in the README (suggested ~25h heartbeat+margin).

**Review by mateo-mro — APPROVED** [review]
